pkg/strategy/tradingdesk/manager.go

```diff
@@ -517,3 +516,119 @@ func fallbackStopLossTakeProfit(
 	}
 	return stopLoss, takeProfit, nil
 }
+
+// extractClosePositionPrice extracts the StopPrice of the first order with ClosePosition set to true.
+func extractClosePositionPrice(orders types.OrderSlice) fixedpoint.Value {
+	for _, order := range orders {
+		if order.ClosePosition {
+			return order.StopPrice
+		}
+	}
+
+	return fixedpoint.Zero
+}
+
+// SlackBlocks generates Slack message blocks with position details, take profit, stop loss prices, and unrealized profit.
+func (m *TradingManager) SlackBlocks() []slack.Block {
+	var blocks []slack.Block
+
+	// Query the latest price
+	ticker, err := m.session.Exchange.QueryTicker(context.Background(), m.Position.Symbol)
```

Comment:
Isn't it better to use a timeout context here?

Reply:
it should be ok, we've set the timeout for http.Client